src/NServiceBus.Core/Recoverability/RecoverabilitySettingsExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using Settings;
+
+    /// <summary>
+    /// Provides extension methods for recoverability settings.
+    /// </summary>
+    public static class RecoverabilitySettingsExtensions
+    {
+        /// <summary>
+        /// Adds the specified exception type to be treated as an unrecoverable exception.
+        /// </summary>
+        /// <param name="settings">The extended settings.</param>
+        /// <param name="exceptionType">The exception type.</param>
+        public static void AddUnrecoverableException(this SettingsHolder settings, Type exceptionType)
+        {
+            Guard.AgainstNull(nameof(settings), settings);
+            Guard.AgainstNull(nameof(exceptionType), exceptionType);
+
+            if (!typeof(Exception).IsAssignableFrom(exceptionType))
+            {
+                throw new ArgumentException("Exception type must be an exception", nameof(exceptionType));
+            }
+
+            HashSet<Type> unrecoverableExceptions;
+            if (!settings.TryGet(Recoverability.UnrecoverableExceptions, out unrecoverableExceptions))
```

Comment:
I know now why I haven't used it. `GetOrCreate `doesn't allow you to specify a key and I thought it is too dangerous to just use `HashSet<Type> `as a key.